crates/uv-resolver/src/lock/mod.rs

```diff
@@ -717,7 +717,10 @@ impl Lock {
     }
 
     /// Returns the TOML representation of this lockfile.
-    pub fn to_toml(&self) -> Result<String, toml_edit::ser::Error> {
+    pub fn to_toml(
+        &self,
+        proxies: Option<&Vec<ProxyWithCanonicalUrl>>,
```

Comment:
I haven't had a chance to review this yet, but as a minor note, I'd expect this to take `Option<&[ProxyWithCanonicalUrl]>` (and similarly for all public APIs). A slice is strictly more flexible than a reference to a `Vec`: if you have a vector, you can always pass it as a slice, but the opposite is not true.